api-reference/10 UI Components/dxScheduler/1 Configuration/appointmentTooltipTemplate.md

```diff
@@ -29,6 +29,399 @@ A template name or container.
+
+    <!-- tab: app.components.ts -->
+    import { TooltipModule } from "./tooltip/tooltip.component";
+
+    @Pipe({ name: "getColor" })
+    class ColorPipe implements PipeTransform {
+        constructor(private service: Service) {}
+        transform(resourceID: number): string {
+            return this.service
+                .getEmployees()
+                .find((employee) => employee.id === resourceID).color;
+        }
+    }
+
+    export class AppComponent {
+        @ViewChild("targetScheduler", { static: true })
+
+        scheduler: DxSchedulerComponent;
+    
+        resourcesDataSource = this.service.getEmployees();
+    
+        constructor(private service: Service) {}
+
+        onDeleteButtonClick(e, appointmentData) {
+            this.scheduler.instance.deleteAppointment(appointmentData);
+            e.event.stopPropagation();
+            this.scheduler.instance.hideAppointmentTooltip();
+        }
+    
+        isDeleteButtonExist(data) {
+            const isDisabled = this.resourcesDataSource.find(
+                (employee) => employee.id === data.appointmentData.employeeID
+            ).disabled;
+            
+            return (
+                !isDisabled &&
+                ((this.scheduler.instance.option("editing") &&
+                this.scheduler.instance.option("editing.allowDeleting") === true) ||
+                this.scheduler.instance.option("editing") === true)
+            );
+        }
+    }
+
+    <!-- tab: tooltip.component.html -->
+    <div class="dx-tooltip-appointment-item">
+        <div class="dx-tooltip-appointment-item-marker">
+            <div
+                class="dx-tooltip-appointment-item-marker-body"
+                [style.background-color]="markerColor" 
+            ></div>
+        </div>
+
+        <div class="dx-tooltip-appointment-item-content">
+            <div class="dx-tooltip-appointment-item-content">
+                <div class="dx-tooltip-appointment-item-content-subject">
+                    {{data.appointmentData.text}}
+                </div>
+                <div class="dx-tooltip-appointment-item-content-date">
+                    {{data.appointmentData.startDate}}
+                </div>
+            </div>
+        </div>
+        <div
+            *ngIf="isDeleteButtonExist"
+            class="dx-tooltip-appointment-item-delete-button-container"
+        >
+            <dx-button
+                class="dx-tooltip-appointment-item-delete-button"
+                icon="trash"
+                stylingMode="text"
+                (onClick)="onDeleteButtonClick.emit($event)"
+            ></dx-button>
+        </div>
+    </div>
+
+    <!-- tab: tooltip.components.ts -->
+        @Component({
+        selector: "Tooltip",
+        templateUrl: "./tooltip.component.html"
+    })
+
+    export class TooltipComponent {
+        @Input() data?: any;
+        @Input() markerColor?: string;
+        @Input() isDeleteButtonExist?: boolean;
```

Comment:
Let's get rid of the ```?``` (because we don't write it in the ```AppComponent```) and change ```data``` input:

```suggestion
        @Input() appointmentData: dxSchedulerAppointment;
        @Input() markerColor: string;
        @Input() isDeleteButtonExist: boolean;
```